src/Init/Data/Order/Factories.lean

```diff
@@ -0,0 +1,279 @@
+  le_min_iff := by simpa [LawfulOrderLE.le_iff] using le_min_iff
+
+/--
+Returns a `LawfulOrderMin α` instance given certain properties.
+
+If an `OrderData α` instance is compatible with an `LE α` instance, then this lemma derives
+a `LawfulOrderMin α` instance from two properties involving `LE α` and `Min α` instances.
+
+This convenience lemma combines `LawfulOrderInf.ofLE` and `LawfulOrderMin.ofLE`.
+-/
+public theorem LawfulOrderMin.ofLE {α : Type u} [OrderData α] [Min α] [LE α] [LawfulOrderLE α]
+    (le_min_iff : ∀ a b c : α, a ≤ Min.min b c ↔ a ≤ b ∧ a ≤ c)
+    (min_eq_or : ∀ a b : α, min a b = a ∨ min a b = b) : LawfulOrderMin α where
+  toLawfulOrderInf := .ofLE le_min_iff
+  toMinEqOr := ⟨min_eq_or⟩
+
+/--
+If an `OrderData α` instance is compatible with an `LE α` instance, then this lemma characterizes
+in terms of `LE α` when a `Max α` instance "behaves like an supremum operator".
+-/
+public def LawfulOrderSup.ofLE {α : Type u} [OrderData α] [Max α] [LE α] [LawfulOrderLE α]
+    (max_le_iff : ∀ a b c : α, Max.max a b ≤ c ↔ a ≤ c ∧ b ≤ c) : LawfulOrderSup α where
+  max_le_iff := by simpa [LawfulOrderLE.le_iff] using max_le_iff
+
+/--
+Returns a `LawfulOrderMax α` instance given certain properties.
+
+If an `OrderData α` instance is compatible with an `LE α` instance, then this lemma derives
+a `LawfulOrderMax α` instance from two properties involving `LE α` and `Max α` instances.
+
+This convenience lemma combines `LawfulOrderSup.ofLE` and `LawfulOrderMax.ofLE`.
+-/
+public def LawfulOrderMax.ofLE {α : Type u} [OrderData α] [Max α] [LE α] [LawfulOrderLE α]
+    (max_le_iff : ∀ a b c : α, Max.max a b ≤ c ↔ a ≤ c ∧ b ≤ c)
+    (max_eq_or : ∀ a b : α, max a b = a ∨ max a b = b) : LawfulOrderMax α where
+  toLawfulOrderSup := .ofLE max_le_iff
+  toMaxEqOr := ⟨max_eq_or⟩
+
+end OfLE
+
+section OfLT
+
+/--
+Creates an `OrderData α` instance from an `LT α` instance.
+
+This only makes sense for asymmetric `LT α` instances (see `Std.Asymm`).
+-/
+public def OrderData.ofLT (α : Type u) [LT α] : OrderData α where
+  IsLE a b := ¬ b < a
+
+/--
+The `OrderData α` instance obtained from an asymmetric `LT α` instance is compatible with said
+`LT α` instance.
+-/
+public instance LawfulOrderLT.ofLT {α : Type u} [LT α] [i : Asymm (α := α) (· < ·)] :
+    haveI := OrderData.ofLT α
+    LawfulOrderLT α :=
+  letI := OrderData.ofLT α
+  { lt_iff a b := by simpa [OrderData.ofLT, Classical.not_not] using i.asymm a b }
+
+/--
+If an `OrderData α` instance is compatible with an asymmetric `LT α` instance the negation of which
+is transitive, then the `OrderData α` instance is a linear order.
+-/
+public theorem IsLinearPreorder.of_lt {α : Type u} [LT α]
+    (lt_asymm : Asymm (α := α) (· < ·) := by exact inferInstance)
+    (not_lt_trans : Trans (α := α) (¬ · < ·) (¬ · < ·) (¬ · < ·) := by exact inferInstance) :
+    haveI : OrderData α := .ofLT α
+    IsLinearPreorder α :=
```

Comment:
Should this not be
```suggestion
public theorem IsLinearPreorder.of_lt {α : Type u} [LT α] [OrderData α] [LawfulOrderLT α]
    (lt_asymm : Asymm (α := α) (· < ·) := by exact inferInstance)
    (not_lt_trans : Trans (α := α) (¬ · < ·) (¬ · < ·) (¬ · < ·) := by exact inferInstance) :
    IsLinearPreorder α :=
```
(simlar for many lemmas below)

Reply:
Okay, I understand. Then this theorem should be called `IsLinearPreorder.ofLT`, no?